lib/helpers/campaign.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const dateFns = require('date-fns');
+const logger = require('winston');
+const config = require('../../config/lib/helpers/campaign');
+
+module.exports = {
+  /**
+   * @param {Object} campaign
+   * @return {Boolean}
+   */
+  isEndDatePast: function isEndDatePast(campaign) {
+    const endDate = dateFns.parse(campaign.endDate.date);
+    return dateFns.isPast(endDate);
+  },
+  /**
+   * @param {Object} campaign
+   * @return {Boolean}
+   */
+  isClosed: function isClosed(campaign) {
+    if (campaign.status) {
+      return this.isClosedStatus(campaign);
+    }
+
+    if (!campaign.endDate) {
+      return true;
+    }
+
+    return this.isEndDatePast(campaign);
+  },
+  isClosedStatus: function isClosedStatus(campaign) {
+    return campaign.status === config.statuses.closed;
+  },
+  /**
+   * @param {Object} campaign
+   * @return {String}
+   */
+  parseStatus: function parseStatus(campaign) {
+    if (this.isClosed(campaign)) {
+      return config.statuses.closed;
+    }
+    return config.statuses.active;
+  },
+  /**
+   * @param {object} campaign
+   * @return {Object}
+   */
+  parseCampaign: function parseCampaign(campaign) {
```

Comment:
Minor suggestion. You could start leaving generous amounts of `//TODO`s for when we switch to use varchar campaign ids.